backend/src/storage/blob.rs

```diff
@@ -52,57 +130,304 @@ impl StorageProvider for BlobStorageProvider {
             })?;
         println!("s_content == {s_content}");
 
-        let flag: BoolFlag = serde_json::from_str(&s_content)
-            .map_err(|e| StorageError::DatabaseError(format!("failed to parse json: {}", e)))?;
+        let flag: BoolFlag = serde_json::from_str(&s_content).map_err(|e| {
+            StorageError::SerializationError(format!("failed to parse json: {}", e))
+        })?;
 
         Ok(Some(flag))
     }
 
-    async fn create_bool_flag(&self, _flag: BoolFlag) -> Result<(), StorageError> {
-        todo!()
+    async fn create_bool_flag(&self, flag: BoolFlag) -> Result<(), StorageError> {
+        if self.get_bool_flag(&flag.id).await?.is_some() {
+            return Err(StorageError::AlreadyExists);
+        }
+        println!("Checking if exists");
```

Comment:
either remove or comment all `println!`s, we'll replace these with logger calls in the future